code/API_definitions/location-retrieval.yaml

```diff
@@ -37,6 +37,10 @@ info:
 
     * **Max Age**: Maximum age of the location information which is accepted for the location retrieval (in seconds).
 
+      * Absence of maxAge means "any age" is acceptable for the client. In other words, this is like maxAge=infinite
+      * maxAge=0 means a fresh calculation is requested by the client.
+      If the system does not support it, or fresh location cannot be calculated at that time for any reason, the API response will be "unknown".

```

Comment:
```suggestion
      If the system does not support it, or fresh location cannot be calculated at that time for any reason, the API response will be "UNKNOWN".

```

Reply:
@jlurien I have an issue here because in location-retrieval we do not have unknown/UNKNOWN. 
We have either 

```
{
  "status": 400,
  "code": "LOCATION_RETRIEVAL.MAXAGE_INVALID_ARGUMENT",
  "message": "maxAge threshold cannot be satisfied"
}
```

or 

```
{
  "status": 404,
  "code": "NOT_FOUND",
  "message": "The specified resource is not found"
}
```

Reply:
400 seems more reasonable to me, as it points explicitly to the maxAge value.

Reply:
Thanks @alpaycetin74 
Works for me and I can update the documentation accordingly.
If not able to provide require freshness (maxAge filled) --> `400` + code `LOCATION_RETRIEVAL.MAXAGE_INVALID_ARGUMENT`
If not able to provide any location for a request without maxAge --> `404`

Reply:
Right, I didn't realize that. 
Regarding errors, to me 400 is for cases when the provided input does not comply with the schema or some constraint that should be known by the client, independently of the device. It could fit for cases where some implementation does not support maxAge=0 for any user (it is still pending how to provide information about limits of the implementation to the client). But for cases where the maxAge cannot be satisfied for a specific device or at a specific moment, I think it is more of a  `404` or even  `422` (the entity combining device and maxAge is not processable by the system).

These RESTful interpretations of HTTP statuses are very subtle anyway.

Reply:
OK - new proposal :)

(1)If  maxAge input does not fit the pattern (like `5fr/"3`)  --> 400
(2)If not able to locate the mobile --> 422 with code LOCATION_RETRIEVAL.UNABLE_TO_PROVIDE_LOCATION
(3)If not able to provide expected freshness -->  422 with a code LOCATION_RETRIEVAL.UNABLE_TO_PROVIDE_MAX_AGE

@jlurien @alpaycetin74

Reply:
For (2) I think 404 is more natural, the (3) is more about the combination of device + maxAge at a given time

(1)If maxAge input does not fit the pattern (like 5fr/"3) --> 400
(2)If not able to locate the mobile --> 404 with code LOCATION_RETRIEVAL.DEVICE_NOT_FOUND
(3)If not able to provide expected freshness --> 422 with a code LOCATION_RETRIEVAL.UNABLE_TO_FULFILL_MAX_AGE

cc @javier-carrocalabor

Reply:
Another difference with location-verification is that in this case we cannot return a lastLocationTime indicating how old is the more recent location

Reply:
Thanks @bigludo7 ,
if I may summarize my view:

For location retrieval,
(1) Obviously a client error, 400 is appropriate. 
It is possible to advise LOCATION_RETRIEVAL.MAXAGE_INVALID_ARGUMENT, but the implementer may not have full control over error messages created in case of syntactic errors - those may be created by software libraries. 
(2) No location available, even if maxAge is absent (infinite). 
404 with code LOCATION_RETRIEVAL.DEVICE_NOT_FOUND is appropriate.
(3) maxAge condition (including 0) cannot be satisfied. 
422 with code LOCATION_RETRIEVAL.UNABLE_TO_FULFILL_MAX_AGE is appropriate.

For location verification,
(1) same
(2) UNKNOWN without lastLocationTime
(3) UNKNOWN, may have  lastLocationTime if system has it.  
Thank you